### ISSUE TITLE
The argument "user.0.groups" is required, but no definition was found

### ISSUE THREAD

**Issue by leetrout** [hashicorp/terraform-provider-vault#1866]
When trying to add an okta user with only policies and no groups defined the plan fails.

### Terraform Version
`1.4.6`

### Affected Resource(s)
Please list the resources as a list, for example:
- vault_okta_auth_backend

### Terraform Configuration Files
```hcl
resource "vault_okta_auth_backend" "lees-test" {
    ...
    user {
        username = "lee@example.com"
        policies   = ["eng", "default"]
    }
}
```

### Expected Behavior
Per the docs a username with policies **without** a group is valid since group is marked as optional but it is acting as it is required.
<img width="597" alt="image" src="https://github.com/hashicorp/terraform-provider-vault/assets/524648/44772b53-e315-4845-9c22-18f5ad8338f3">

### Actual Behavior
Plan failed.

```
The argument "user.0.groups" is required, but no definition was found.
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

**Comment by raymonstah** [comment]
Hi @leetrout thanks for raising this issue! I've put up a fix for it at #1902. I'll circle back and let you know when it has been released.

Thanks again!

**Comment by leetrout** [comment]
> Hi @leetrout thanks for raising this issue! I've put up a fix for it at #1902. I'll circle back and let you know when it has been released.
> 
> 
> 
> Thanks again!

Thank you!

**Comment by raymonstah** [comment]
Just wanted to let you know that this was released yesterday as part of [v3.17.0](https://github.com/hashicorp/terraform-provider-vault/releases/tag/v3.17.0).
